python/TTModule.cpp

```diff
@@ -301,9 +302,14 @@ void populateTTModule(py::module &m) {
               chipChannelsUnwrapped.push_back(
                   mlir::cast<tt::ChipChannelAttr>(unwrap(chipChannel)));
             }
+            std::vector<tt::CPUDescAttr> cpuDescsUnwrapped;
+            for (const auto &cpuDesc : cpuDescs) {
```

Comment:
similarly, I think the rest of these loops should be `const auto &` not just `auto`

Reply:
Good catches... I think a seperate PR would be very nice to cleanup other offenders (if you're inclined afterwards).